src/librustdoc/json/conversions.rs

```diff
@@ -0,0 +1,598 @@
+//! These from impls are used to create the JSON types which get serialized. They're very close to
+//! the `clean` types but with some fields removed or stringified to simplify the output and not
+//! expose unstable compiler internals.
+
+use std::convert::From;
+
+use rustc_ast::ast;
+use rustc_span::def_id::{DefId, CRATE_DEF_INDEX};
+
+use crate::clean;
+use crate::doctree;
+use crate::formats::item_type::ItemType;
+use crate::json::types::*;
+
+impl From<clean::Item> for Option<Item> {
+    fn from(item: clean::Item) -> Self {
+        let item_type = ItemType::from(&item);
+        let clean::Item {
+            source,
+            name,
+            attrs,
+            kind,
+            visibility,
+            def_id,
+            stability: _,
+            deprecation,
+        } = item;
```

Comment:
This failed to compile together with other approved PRs, in https://github.com/rust-lang/rust/pull/79595: https://github.com/rust-lang/rust/runs/1479419205?check_suite_focus=true

Reply:
Specifically, this one: https://github.com/rust-lang/rust/pull/79548

Reply:
@m-ou-se there's no fix for that, it's a semantic conflict. One or the other will have to be merged first (my preference is for this one, but it shouldn't really matter).

Reply:
Sure, just a heads up.

Reply:
Fixed in 91e9ae0. Does bors need to see approval again?

Reply:
Yes, I re-approved.